### ISSUE TITLE
Some accessibility issues

### ISSUE THREAD

**Issue by mgifford** [squidfunk/mkdocs-material#6433]
### Context

We were looking at the accessibility of our site which uses your theme - https://github.com/CivicActions/guidebook/issues/1307

I think some of them are yours. 

### Bug description

This needs to be a link - https://github.com/squidfunk/mkdocs-material/blob/0b38323a310d9ab9ac7fd60ad0bdae5871e42d73/src/templates/partials/search.html#L100

Although this works with simple keyboard only testing (thanks to JS) it won't be focusable if the JS is disabled. 

```
          <div class="md-search-result__meta">
            {{ lang.t("search.result.initializer") }}
          </div>
```

Probably should just be:

          <a class="md-search-result__meta">
            {{ lang.t("search.result.initializer") }}
          </a>
          
if JS is enabled, then you can dynamically change the form to the larger label for folks to enter in the search element.  If not, the link could actually take them to a search page. 

For screen reader users, when content dynamically changes, you also need to announce the change using aria-live https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live

There is likely going to be more work that needs to be done here for when results are dynamically returned to the user. 

Currently, axe reports this on our site: 

Title: WCAG 2.1.1: Ensure elements that have scrollable content are accessible by keyboard (.md-search__scrollwrap)
Tags: Accessibility, WCAG 2.1.1, scrollable-region-focusable

Issue: Ensure elements that have scrollable content are accessible by keyboard (scrollable-region-focusable - https://accessibilityinsights.io/info-examples/web/scrollable-region-focusable)

Target application: CivicActions Guidebook - CivicActions Guidebook - https://guidebook.civicactions.com/en/latest/

Element path: .md-search__scrollwrap

Snippet: <div class="md-search__scrollwrap"> <div class="md-search-result" data-md-component="search-result"> <div class="md-search-result__meta">Type to start searching</div> <ol class="md-search-result__list" role="presentation"></ol> </div> </div>

How to fix: 
Fix any of the following:
  Element should have focusable content
  Element should be focusable

Environment: Microsoft Edge version 119.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.41.0 (axe-core 4.7.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live
- https://github.com/squidfunk/mkdocs-material/blob/0b38323a310d9ab9ac7fd60ad0bdae5871e42d73/src/templates/partials/search.html#L100
- https://accessibilityinsights.io/info-examples/web/scrollable-region-focusable
- https://web.dev/learn/css/focus/

### Reproduction

Go to a site with this search and leverage this tool http://aka.ms/AccessibilityInsights

### Steps to reproduce

Fastest way is to go to https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug and run axe or accessibility insights.  There are lots of accessibility issues with your demo page. 

FastPass_20231204_ReportingabugMateria.html.zip

### Browser

Edge

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
> Although this works with simple keyboard only testing (thanks to JS) it won't be focusable if the JS is disabled.

> if JS is enabled, then you can dynamically change the form to the larger label for folks to enter in the search element. If not, the link could actually take them to a search page.

Search only works with JS. Without JS, there is no search, because search is entirely client side. This theme does not have a dedicated search page, and if it had, it would also only work with JS.

> This needs to be a link

The element only shows non-actionable content, i.e., there's nothing to click. It shows "Type to start searching" and when you search "38 documents found". Thus, it doesn't need to be a link, because there's nothing to link to.
